pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -687,6 +739,25 @@ func (ctrl *Controller) syncImageConfig(key string) error {
 	} else if err != nil {
 		return err
 	}
+	// Find all ImageDigestMirrorSet objects
+	idmsRules, err := ctrl.idmsLister.List(labels.Everything())
+	if err != nil && errors.IsNotFound(err) {
+		idmsRules = []*apicfgv1.ImageDigestMirrorSet{}
+	} else if err != nil {
+		return err
+	}
+
+	// Find all ImageTagMirrorSet objects
+	itmsRules, err := ctrl.itmsLister.List(labels.Everything())
+	if err != nil && errors.IsNotFound(err) {
+		itmsRules = []*apicfgv1.ImageTagMirrorSet{}
+	} else if err != nil {
+		return err
+	}
+
+	if err := runtimeutils.RejectMultiUpdateMirrorSetObjs(icspRules, idmsRules, itmsRules); err != nil {
```

Comment:
Without getting in https://github.com/openshift/kubernetes/pull/1310, the user will not detect the error until they look at the logs.
I think with https://github.com/openshift/kubernetes/pull/1310  stopping updating different crds from the API level, theoretically, it should fail before getting here.
What https://github.com/openshift/kubernetes/pull/1310 did has been discussed https://docs.google.com/document/d/13h6IJn8wlzXdiPMvCWlMEHOXXqEZ9_GYOl02Wldb3z8/edit#heading=h.shj0ebufp0yj, section Questions 1.c. I hope it can pass the ci

Reply:
ack, thanks!